packages/sommelier.rb

```diff
@@ -47,246 +39,281 @@ class Sommelier < Package
     @peer_cmd_prefix = '/lib64/ld-linux-x86-64.so.2'
   end
 
+  def self.preflight
+    unless File.socket?('/var/run/chrome/wayland-0')
+      abort "This package is not compatible with your device :/".lightred
+    end
+  end
+
+  def self.prebuild
+    # Patch to avoid error with GCC > 9.x
+    # ../sommelier.cc:3238:10: warning: ‘char* strncpy(char*, const char*, size_t)’ specified bound 108 equals destination size [-Wstringop-truncation]
+    Kernel.system "sed -i 's/sizeof(addr.sun_path))/sizeof(addr.sun_path) - 1)/' sommelier.cc", chdir: 'vm_tools/sommelier'
```

Comment:
Is there an advantage to using `Kernel.system` over just `system`?

( Asking because I have no idea! ) :)

Reply:
It is for bypassing the patched `system` function in `package.rb`, The `package.rb` one won't work as it will throw an error when trying the `chdir` hash

Reply:
@uberhacker Any suggestions on making `def self.system(*args)` in `package.rb` work for this?